Mathlib/Geometry/Manifold/Instances/Real.lean

```diff
@@ -245,6 +245,33 @@ def IccLeftChart (x y : ℝ) [h : Fact (x < y)] :
     have B : Continuous fun z : EuclideanSpace ℝ (Fin 1) => z 0 := continuous_apply 0
     exact (A.comp B).comp continuous_subtype_val
 
+variable {x y : ℝ} [hxy : Fact (x < y)]
+
+/-- The endpoint `x ∈ Icc x y`, as a point in `Icc x y` (assuming `x < y`). -/
+abbrev IccManifold.left : Icc x y := ⟨x, ⟨le_refl x, by have := hxy.out; linarith⟩⟩
+
+/-- The endpoint `y ∈ Icc x y`, as a point in `Icc x y` (assuming `x < y`). -/
+abbrev IccManifold.right : Icc x y := ⟨y, ⟨by have := hxy.out; linarith, le_refl y⟩⟩
```

Comment:
This has nothing to do with the manifold setup. Shouldn't this be available to anyone working with the set `Icc`?

Reply:
Good point: I moved this to the definition of Icc. The names `Icc.left` and `Icc.right` might cause conflicts; if you prefer a different name, I'm open to changing it.